src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Properties/InterceptedProjectProperties/LocalDebuggerCommandArgumentsValueProvider.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.VisualStudio.ProjectSystem.Build;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using System;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Properties
+{
+    [ExportInterceptingPropertyValueProvider("LocalDebuggerCommandArguments", ExportInterceptingPropertyValueProviderFile.UserFileWithXamlDefaults)]
+    internal sealed class LocalDebuggerCommandArgumentsValueProvider : InterceptingPropertyValueProviderBase
+    {
+        private readonly Lazy<ProjectProperties> _projectProperties;
+        private readonly Lazy<IOutputGroupsService, IAppliesToMetadataView> _outputGroups;
+        private readonly ConfiguredProject _configuredProject;
+
+        [ImportingConstructor]
+        public LocalDebuggerCommandArgumentsValueProvider(Lazy<ProjectProperties> projectProperties, 
+                                                          [Import(AllowDefault = true)] Lazy<IOutputGroupsService, IAppliesToMetadataView> outputGroups,
+                                                          ConfiguredProject configuredProject)
+        {
+            Requires.NotNull(projectProperties, nameof(projectProperties));
+            Requires.NotNull(configuredProject, nameof(configuredProject));
+
+            _projectProperties = projectProperties;
+            _outputGroups = outputGroups;
+            _configuredProject = configuredProject;
+        }
+
+        private Lazy<IOutputGroupsService> OutputGroups {
+            get {
+                return _outputGroups != null && _outputGroups.AppliesTo(_configuredProject.Capabilities) ? _outputGroups : null;
```

Comment:
I'm wondering if we want this to be the OrderPrecedenceColletion<T>, it handles this already - and avoids the race you have below where capabilities can change underneath you (you're guaranteed that a single call "context" for the capabilities to remain consistent)